StyleCop.Analyzers/StyleCop.Analyzers/OrderingRules/SA1209UsingAliasDirectivesMustBePlacedAfterOtherUsingDirectives.cs

```diff
@@ -47,64 +51,67 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+        }
 
-            if (!usingDirectives.HasValue)
-            {
-                return;
-            }
+        private void HandleNamespaceDeclaration(SyntaxNodeAnalysisContext context)
+        {
+            var namespaceDeclaration = context.Node as NamespaceDeclarationSyntax;
+
+            ProcessUsingsAndReportDiagnostic(namespaceDeclaration.Usings, context);
+        }
+
+        private static void ProcessUsingsAndReportDiagnostic(SyntaxList<UsingDirectiveSyntax> usings, SyntaxNodeAnalysisContext context)
+        {
+            UsingDirectiveSyntax usingAliasDirectivesShouldBePlacedAfterThis = null;
+            var usingAliasDirectivesToReport = new Lazy<List<UsingDirectiveSyntax>>();
 
-            bool foundCurrent = false;
-            foreach (var usingDirective in usingDirectives)
+            for (int i = 0; i < usings.Count; i++)
             {
-                // we are only interested in nodes after the current node
-                if (usingDirective == syntax)
-                {
-                    foundCurrent = true;
-                    continue;
-                }
-                else if (!foundCurrent)
+                var usingDirective = usings[i];
+                var notLastUsingDirective = i + 1 < usings.Count;
+                if (usingDirective.Alias != null && notLastUsingDirective)
                 {
-                    continue;
+                    var nextUsingDirective = usings[i + 1];
+                    if (nextUsingDirective.Alias == null && nextUsingDirective.StaticKeyword.IsKind(SyntaxKind.None))
+                    {
+                        usingAliasDirectivesToReport.Value.Add(usingDirective);
+                    }
                 }
-
-                // ignore following using alias directives
-                if (usingDirective.Alias != null)
+                else
                 {
-                    continue;
+                    usingAliasDirectivesShouldBePlacedAfterThis = usingDirective;
                 }
+            }
 
-                SymbolInfo symbolInfo = context.SemanticModel.GetSymbolInfo(usingDirective.Name, context.CancellationToken);
-                INamespaceSymbol followingNamespaceSymbol = symbolInfo.Symbol as INamespaceSymbol;
-                if (followingNamespaceSymbol == null)
-                {
-                    continue;
+            if (usingAliasDirectivesToReport.IsValueCreated && usingAliasDirectivesToReport.Value.Count > 0)
+            {
+                var unaliasedNamespaceName = GetNamespaceNameWithoutAlias(usingAliasDirectivesShouldBePlacedAfterThis.Name.ToString());
+                foreach (var usingAliasDirectiveToReport in usingAliasDirectivesToReport.Value)
+                {                    
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, usingAliasDirectiveToReport.GetLocation(), usingAliasDirectiveToReport.Alias.Name.ToString(), unaliasedNamespaceName));
                 }
+            }
+        }
```

Comment:
If the list of usings ends with several aliased usings, the message will suggest to place the reported aliased using (IO in the example) after the last aliased using statement (`System.Threading` in the example), instead of the alphabetically correct location (`System.Net` in the example).

Example:
```C#
using System;
using IO = System.IO;
using System.Net;
using Numerics = System.Numerics;
using Threads = System.Threading;
```